web/event_utils.js

```diff
@@ -185,8 +185,11 @@ class AutomationEventBus extends EventBus {
         details[key] = value;
       }
     }
-    const event = document.createEvent("CustomEvent");
-    event.initCustomEvent(eventName, true, true, details);
+    const event = new CustomEvent(eventName, {
+      bubbles: true,
+      cancelable: true,
+      detail: details,
```

Comment:
Nit: I'd probably just rename https://github.com/mozilla/pdf.js/pull/16336/files#diff-f2eaa64e443dc3c28c4f728ae3cef8e07569df4cf9a5d244dd9b892e2f45656fR175 to `const detail = Object.create(null);` and then write this as follows.
```suggestion
      detail,
```

Reply:
Sounds good; will do.